# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1502802167 14400
#      Tue Aug 15 09:02:47 2017 -0400
# Node ID 7662b441e2eabded0a7c61947707700c84456f6d
# Parent  47c44b822d114fa4e4b5db4faa2b68d74fd90d1e
eliminate unused Octave:undefined-return-values warning ID

* NEWS: Note that Octave:undefined-return-values is obsolete.

* warning_ids.m: Delete Octave:undefined-return-values from the list.

* dos.m, unix.m: Don't set Octave:undefined-return-values warning ID
in tests.

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -72,16 +72,20 @@ Summary of important user-visible change
       nfields
       octave_tmp_file_name
       playaudio
       saveaudio
       setaudio
       syl
       usage
 
+ ** The "Octave:undefined-return-values" warning ID is obsolete.  Octave
+    now throws an error for any attempts to assign undefined values that
+    might be returned from functions.
+
  ** Deprecated graphics properties.
 
     The following properties or allowed corresponding values have been
     deprecated in Octave 4.4 and will be removed from Octave 4.8 (or whatever
     version is the second major release after 4.4):
 
       Object               | Property                | Value
       ---------------------|-------------------------|-------------------
diff --git a/scripts/help/warning_ids.m b/scripts/help/warning_ids.m
--- a/scripts/help/warning_ids.m
+++ b/scripts/help/warning_ids.m
@@ -329,22 +329,16 @@
 ## @result{} 97 98 99
 ## @end group
 ## @end example
 ##
 ## @noindent
 ## elicits a warning if the @code{Octave:str-to-num} warning is enabled.
 ## By default, the @code{Octave:str-to-num} warning is disabled.
 ##
-## @item Octave:undefined-return-values
-## If the @code{Octave:undefined-return-values} warning is disabled,
-## print a warning if a function does not define all the values in
-## the return list which are expected.
-## By default, the @code{Octave:undefined-return-values} warning is enabled.
-##
 ## @item Octave:variable-switch-label
 ## If the @code{Octave:variable-switch-label} warning is enabled, Octave
 ## will print a warning if a switch label is not a constant or constant
 ## expression.
 ## By default, the @code{Octave:variable-switch-label} warning is disabled.
 ## @end table
 ##
 
diff --git a/scripts/miscellaneous/dos.m b/scripts/miscellaneous/dos.m
--- a/scripts/miscellaneous/dos.m
+++ b/scripts/miscellaneous/dos.m
@@ -48,17 +48,16 @@ function [status, text] = dos (command, 
     endif
   endif
 
 endfunction
 
 
 %!test
 %! cmd = ls_command ();
-%! warning ("off", "Octave:undefined-return-values", "local");
 %! [status, output] = dos (cmd);
 %!
 %! if (ispc () && ! isunix ())
 %!   [status, output] = dos (cmd);
 %!   assert (status, 0);
 %!   assert (ischar (output));
 %!   assert (! isempty (output));
 %! else
diff --git a/scripts/miscellaneous/unix.m b/scripts/miscellaneous/unix.m
--- a/scripts/miscellaneous/unix.m
+++ b/scripts/miscellaneous/unix.m
@@ -46,17 +46,16 @@ function [status, text] = unix (command,
     endif
   endif
 
 endfunction
 
 
 %!test
 %! cmd = ls_command ();
-%! warning ("off", "Octave:undefined-return-values", "local");
 %! [status, output] = unix (cmd);
 %!
 %! if (isunix ())
 %!   assert (status, 0);
 %!   assert (ischar (output));
 %!   assert (! isempty (output));
 %! else
 %!   assert (status, []);
